src/constants.js

```diff
@@ -0,0 +1,214 @@
+export const GRANULARITY_OPTIONS = {
+  LOW: 'low',
+  MEDIUM: 'medium',
+  HIGH: 'high',
+  AUTO: 'auto',
+  DENSE: 'dense',
+  CUSTOM: 'custom'
+};
+
+export const TARGETING_KEYS = {
+  BIDDER: 'hb_bidder',
+  AD_ID: 'hb_adid',
+  PRICE_BUCKET: 'hb_pb',
+  SIZE: 'hb_size',
+  DEAL: 'hb_deal',
+  SOURCE: 'hb_source',
+  FORMAT: 'hb_format',
+  UUID: 'hb_uuid',
+  CACHE_ID: 'hb_cache_id',
+  CACHE_HOST: 'hb_cache_host',
+  ADOMAIN: 'hb_adomain',
+  ACAT: 'hb_acat',
+  CRID: 'hb_crid',
+  DSP: 'hb_dsp'
+};
+
+export const DEFAULT_TARGETING_KEYS = {
+  BIDDER: 'hb_bidder',
+  AD_ID: 'hb_adid',
+  PRICE_BUCKET: 'hb_pb',
+  SIZE: 'hb_size',
+  DEAL: 'hb_deal',
+  FORMAT: 'hb_format',
+  UUID: 'hb_uuid',
+  CACHE_HOST: 'hb_cache_host'
+};
+
+export const NATIVE_KEYS = {
+  title: 'hb_native_title',
+  body: 'hb_native_body',
+  body2: 'hb_native_body2',
+  privacyLink: 'hb_native_privacy',
+  privacyIcon: 'hb_native_privicon',
+  sponsoredBy: 'hb_native_brand',
+  image: 'hb_native_image',
+  icon: 'hb_native_icon',
+  clickUrl: 'hb_native_linkurl',
+  displayUrl: 'hb_native_displayurl',
+  cta: 'hb_native_cta',
+  rating: 'hb_native_rating',
+  address: 'hb_native_address',
+  downloads: 'hb_native_downloads',
+  likes: 'hb_native_likes',
+  phone: 'hb_native_phone',
+  price: 'hb_native_price',
+  salePrice: 'hb_native_saleprice',
+  rendererUrl: 'hb_renderer_url',
+  adTemplate: 'hb_adTemplate'
+};
+
+export const S2S = {
+  SRC: 's2s',
+  DEFAULT_ENDPOINT: 'https://prebid.adnxs.com/pbs/v1/openrtb2/auction',
+  SYNCED_BIDDERS_KEY: 'pbjsSyncs'
+};
+
+export const BID_STATUS = {
+  BID_TARGETING_SET: 'targetingSet',
+  RENDERED: 'rendered',
+  BID_REJECTED: 'bidRejected'
+};
+
+export const REJECTION_REASON = {
+  INVALID: 'Bid has missing or invalid properties',
+  INVALID_REQUEST_ID: 'Invalid request ID',
+  BIDDER_DISALLOWED: 'Bidder code is not allowed by allowedAlternateBidderCodes / allowUnknownBidderCodes',
+  FLOOR_NOT_MET: 'Bid does not meet price floor',
+  CANNOT_CONVERT_CURRENCY: 'Unable to convert currency',
+  DSA_REQUIRED: 'Bid does not provide required DSA transparency info',
+  DSA_MISMATCH: 'Bid indicates inappropriate DSA rendering method'
+};
+
+export const PREBID_NATIVE_DATA_KEYS_TO_ORTB = {
+  body: 'desc',
+  body2: 'desc2',
+  sponsoredBy: 'sponsored',
+  cta: 'ctatext',
+  rating: 'rating',
+  address: 'address',
+  downloads: 'downloads',
+  likes: 'likes',
+  phone: 'phone',
+  price: 'price',
+  salePrice: 'saleprice',
+  displayUrl: 'displayurl'
+};
+
+export const NATIVE_ASSET_TYPES = {
+  sponsored: 1,
+  desc: 2,
+  rating: 3,
+  likes: 4,
+  downloads: 5,
+  price: 6,
+  saleprice: 7,
+  phone: 8,
+  address: 9,
+  desc2: 10,
+  displayurl: 11,
+  ctatext: 12
+};
+
+export const NATIVE_IMAGE_TYPES = {
+  ICON: 1,
+  MAIN: 3
+};
+
+export const NATIVE_KEYS_THAT_ARE_NOT_ASSETS = [
+  'privacyIcon',
+  'clickUrl',
+  'sendTargetingKeys',
+  'adTemplate',
+  'rendererUrl',
+  'type'
+];
+
+export const FLOOR_VALUES = {
```

Comment:
this should be in `pubmaticAnalyticsAdapter`

Reply:
Move :truck: